qiskit_dynamics/pulse/pulse_to_signals.py

```diff
@@ -65,14 +92,17 @@ def get_signals(self, schedule: Schedule) -> List[DiscreteSignal]:
             qiskit.QiskitError: if not enough frequencies supplied
         """
 
-        if self._carriers and len(self._carriers) < len(schedule.channels):
-            raise QiskitError("Not enough carrier frequencies supplied.")
+        if self._channels is None:
+            if self._carriers and len(self._carriers) < len(schedule.channels):
+                raise QiskitError("Not enough carrier frequencies supplied.")
 
```

Comment:
Can you add a comment here saying what data is being initialized - these dicts and the subsequent loop are nearly self explanatory but would be helpful to have a brief comment on how these dicts will be used.

Reply:
The changes here c37d941 simplify the interface quite a bit. This also reduces the documentation need.